lisa/node.py

```diff
@@ -548,8 +550,17 @@ def _execute(
             update_envs=update_envs,
             command_splitter=command_splitter,
         )
+        self._check_process_password(process)
         return process
 
+    def _check_process_password(self, process: Process) -> None:
+        timer = create_timer()
+        while process.is_running():
+            time.sleep(0.01)
+            if timer.elapsed(False) > 0.5:
```

Comment:
The check logic should be optimized to avoid wait 0.5 second for every command. It impacts the overall performance.
1. Check immediately one time. Please look at the logic inside of check_and_input_password, there are 3 conditions. If no condition met, it should skip following checks.
2. change the timer to 0.2, see if it still works.

Reply:
that's a good suggestion. I kept this logic here as the change doesn't change the original behavior. As to optimization, I tried to call process.check_and_input_password() without waiting. It works from my test, but I am not sure it covers all the situations. I will update the change for your review.

Reply:
You can check with Baihua, how 0.5 second is from.

Reply:
After discussion with Baihua, we took your first suggestion and made appropriate changes for check_and_input_password() for safety reason. 0.5 wait time can be changed to 0.2 from my test, but Baihua preferred to keep it to avoid regression. Please let me know if you have further suggestions.